appmap/command/appmap_agent_validate.py

```diff
@@ -23,31 +23,37 @@ def check_python_version():
     except AppMapPyVerException as e:
         raise ValidationFailure(str(e))
 
-def _check_version(dist, v):
+def _check_version(dist, versions):
     dist_version = None
-    try:
-        dist_version = version(dist)
+    for v in versions:
+        try:
+            dist_version = version(dist)
+
+            required = parse(v)
+            actual = parse(dist_version)
+            if required.major != actual.major:
+                dist_version = None
+                continue
 
-        required = parse(v)
-        actual = parse(dist_version)
+            if actual < required:
+                raise ValidationFailure(f'{dist} must have version >= {required}, found {actual}')
 
-        if actual < required:
-            raise ValidationFailure(f'{dist} must have version >= {required}, found {actual}')
-    except PackageNotFoundError:
-        pass
+            return dist_version
+        except PackageNotFoundError:
+            dist_version = None
```

Comment:
Nitpick: this changed logic is a bit convoluted and difficult to follow. How about something like:
```python
try:
  dist_version = version(dist)
  actual = parse(dist_version)
  required = next(v for v in map(parse, versions) if v.major == actual.major)
  if actual < required:
    raise ValidationFailure
except PackageNotFoundError, StopIteration:
  return None
return dist_version
```